### PR TITLE
Add framed queries & refactoring operations

### PR DESCRIPTION
**Pull request by tbolier**

### CODE REVIEW

File: src/Connection/Socket/Handshake.php [hunk 1]
```diff
@@ -282,15 +280,32 @@ private function verifyAuthentication($response): string
...
 
         $this->state = 4;
 
         return 'successful';
     }
+
+    /**
+     * @param null|string $handshakeResponse
+     */
+    private function checkResponse(?string $handshakeResponse)
```
Comment (AxaliaN) [inline src/Connection/Socket/Handshake.php]: missing void return type

File: src/Connection/Socket/Handshake.php [hunk 2]
```diff
@@ -282,15 +280,32 @@ private function verifyAuthentication($response): string
...
+                $errorMatch)) {
+            throw new Exception($errorMatch[1]);
+        }
+    }
+
+    /**
+     * @param string $signature
+     * @throws Exception
+     */
+    private function checkSignature(string $signature)
```
Comment (AxaliaN) [inline src/Connection/Socket/Handshake.php]: missing void return type

File: src/Query/Operation/AbstractOperation.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+namespace TBolier\RethinkQL\Query\Operation;
+
+use TBolier\RethinkQL\Query\AbstractQuery;
+
+abstract class AbstractOperation extends AbstractQuery
+{
+    /**
+     * @inheritdoc
+     */
+    public function count(): AbstractOperation
```
Comment (AxaliaN) [inline src/Query/Operation/AbstractOperation.php]: Wouldn't it be nicer if these would have an Interface as return type? Nitpicking maybe ;)

Comment (tbolier) [reply src/Query/Operation/AbstractOperation.php]: Done and correctly set the return types based on the right interface.

**Pull request merged**